integration_testing/connection_controller_routines_test.go

```diff
@@ -302,6 +302,49 @@ func TestConnectionControllerValidatorInboundDeduplication(t *testing.T) {
 	t.Logf("Test #1 passed | Successfully run non-validator node1; validators node2, node3 with duplicate public key")
 }
 
+func TestConnectionControllerNonValidatorConnector(t *testing.T) {
+	require := require.New(t)
+
+	// Spawn 6 non-validators node1, node2, node3, node4, node5, node6. Set node1's targetOutboundPeers to 3. Then make
+	// node1 create persistent outbound connections to node2, node3, and node4, as well as non-validator connections to
+	// node5 and node6.
+	node1 := spawnNonValidatorNodeProtocol2(t, 18000, "node1")
+	node1.Config.TargetOutboundPeers = 0
+	node2 := spawnNonValidatorNodeProtocol2(t, 18001, "node2")
+	node3 := spawnNonValidatorNodeProtocol2(t, 18002, "node3")
+	node4 := spawnNonValidatorNodeProtocol2(t, 18003, "node4")
+	node5 := spawnNonValidatorNodeProtocol2(t, 18004, "node5")
+	node6 := spawnNonValidatorNodeProtocol2(t, 18005, "node6")
+
+	node2 = startNode(t, node2)
+	defer node2.Stop()
```

Comment:
Your `startNode` function already cleans these up at the end of the test. We can get rid of all the `node.Stop` calls here.

Reply:
Lol forgot about it. Nice find!